packages/errors/src/errorTypes.ts

```diff
@@ -0,0 +1,50 @@
+import type { AllCypressErrors } from './errors'
+
+/**
+ * An "error like" is an object which may or may-not be an error,
+ * but contains at least a name & message, and probably a stack
+ */
+export interface ErrorLike {
+  name: string
+  message: string
+  stack?: string
+  // An error-like can have additional properties
+  [key: string]: any
+}
+
+/**
+ * An error originating from the @cypress/errors package,
+ * includes the `type` of the error, the `originalError`
+ * if one exists, and an isCypressError for duck-type checking
+ */
+export interface CypressError extends ErrorLike {
+  type: keyof typeof AllCypressErrors
+  isCypressErr: true
+  originalError?: CypressError | ErrorLike
+  details?: string
+  code?: string | number
+  errno?: string | number
+  stackWithoutMessage?: string
+  forBrowser: ErrTemplateResult['forBrowser']
+}
+
+export interface ErrTemplateResult {
+  message: string
+  details?: string
```

Comment:
now that we cleaned up the normalization of the properties shouldn't these 2 properties be put under like a `forTerminal()` function? this would match the graphql schema better

Reply:
Since the `message` and `details` top level are only used for the terminal - whereas the `forBrowser()` returned `message` and `details` are only used for the browser

Reply:
From slack convo with tim:

> Yeah I was thinking that might be part of the 10 PR rather than adding more surface area to this one, it would be a massive change and the 9.x app doesn't even use forBrowser right now
>
> Bc it does ansi stripping in the browser or something